### PR TITLE
Add `yii\web\Request::getHostName()`

### LINKED ISSUE #11309: Request class cann't return ONLY domain (without scheme)
`Yii::$app->request->getHostInfo()` returns 'http://example.com'. There is no method for handling $_SERVER['HTTP_HOST'] and returning ONLY domain, e.g. `Yii::$app->request->getDomain()` which returns only 'example.com'.
I think will be better to add this method in yii\webRequest class for working in same yii-interface with superglobal variable $_SERVER.

### PR DESCRIPTION
**Pull request by rob006**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | no |
| Tests pass? | yes |
| Fixed issues | #11309 |

fixes #11309
replaces #12670

### CODE REVIEW

**Review by samdark — APPROVED** [review]

File: framework/web/Request.php [hunk 1]
```diff
@@ -554,13 +555,29 @@ public function getHostInfo()
...
     {
+        $this->_hostName = null;
         $this->_hostInfo = $value === null ? null : rtrim($value, '/');
     }
 
+    /**
+     * Returns the host part of the current request URL.
+     * Value is calculated from current [[getHostInfo()|hostInfo]] property.
+     * @return string|null hostname part of the request URL (e.g. `www.yiiframework.com`)
+     * @see getHostInfo()
```
Comment (cebe) [inline framework/web/Request.php]: need `@since 2.0.10`

**Review by cebe — CHANGES_REQUESTED** [review]
looks great, except the `@since` annotation.

File: framework/web/Request.php (hunk 1 above)
Comment (rob006) [inline framework/web/Request.php]: Done.

**Review by cebe — APPROVED** [review]
ready to be merged for 2.0.10 imo.

**Comment by samdark** [comment]
Thank you!

**Pull request merged**